service/dap/server.go

```diff
@@ -755,10 +755,12 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 	// TODO(polina): Respond with an error if debug session is in progress?
 	program, ok := request.Arguments["program"].(string)
 	if !ok || program == "" {
-		s.sendErrorResponse(request.Request,
-			FailedToLaunch, "Failed to launch",
-			"The program attribute is missing in debug configuration.")
-		return
+		if !(mode == "replay" || mode == "core") { // Only fail on modes requiring a program
```

Comment:
Doesn't core require it?
```
$ dlv core
Error: you must provide a core file and an executable
```

Reply:
I'll verify that tomorrow (currently on macOS). rr is maped to the source but its a completely different backend that core so I'm inclined to believe that it does though

Reply:
It seems that program should be expected on all scenarios regardless of the mode and usage since dap expects it

From vs code, it will always arrive, since [this validation](https://github.com/golang/vscode-go/blob/master/src/goDebugFactory.ts#L520-L522) assures it before spawning the dap process and if I comment this validation I get the following error from the service either way (from both replay and core):

![Screenshot from 2021-06-29 20-30-42](https://user-images.githubusercontent.com/6326271/123881120-73328580-d91a-11eb-8224-0055a16fe47c.png)

So I think I can remove the mode exclusions and go back to validate expecting program always, what do you think?

Reply:
We control the extension code. It requires program attribute because up until now it was needed for all launch modes. If we support additional modes that don't need it, it would make sense to remove that requirement.

You don't have unittests, but have you run both repaly and coredump manually via dlv-dap? Have you tried passing a value to program that would be a valid directory, but not a valid executable. Does coredump still work?

Reply:
I ran said tests and core mode requires a program to work so I removed it from the mode exclusions. Having said so, replay works without a program but with the catch you just mentioned, it still needs to have a program arg pointing to some valid location in order to be able to continue because of the validation that seems to happen before onLaunchRequest (otherwise it will throw the error of my previous comment)

Reply:
I believe you have another PR on vscode-go side to enable what you are adding here. In that PR, you can change the program requirement not to apply to the new replay mode to avoid the unnecessary error.

Reply:
I got it even by commenting the check so I think the error is coming from somewhere else in the extension code on on the server side, just no from onLaunchRequest it seems. It definitely warrants some more research, although for all useful purposes it does not constitute a behavior that makes the feature unusable